### PR TITLE
Release review

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request contains the following changes:
* The install directory tree has been modified in order to be compliant with eProsima standard.
* SOVERSION property has been added to CMake.
* Duplicated code has been removed from CMakeLists.txt.
* Installer naming has been update to the new product name.
* Fast CDR has been updated.
* A new CMake packaging test has been added.

### CODE REVIEW

File: microxrce_agent.cpp [hunk 1]
```diff
@@ -203,11 +195,52 @@ int main(int argc, char** argv)
...
+            initialized = true;
+        }
+        std::cout << ((!initialized) ? "ERROR" : "OK") << std::endl;
+    }
+    else
+    {
+        std::cout << "Error server type" << std::endl;
+    }
+
+#endif
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply microxrce_agent.cpp]: I did that for avoid extra complexity.
Firstly Windows servers do not support discovery.
Secondly Windows installer create a .exe in which you can not introduce argument, so the behavior should be different.
For the near future is planning the add a CLI similar to [CLI11](https://github.com/CLIUtils/CLI11), in order to unify the MicroXRCEAgent behavior.

File: test/cmake/main.cpp [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        while ('q' != exit_flag)
+        {
+            std::cout << "Enter 'q' for exit" << std::endl;
+            std::cin >> exit_flag;
+        }
+    }
+
+    return 0;
+}
+
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply test/cmake/main.cpp]: It is only for CMake packaging testing purposes, in fact this file could be simplify.
I will update it.

File: microxrce_agent.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (julianbermudez) [reply microxrce_agent.cpp]: I followed your advice and took common factor for Linux and Windows.

File: test/cmake/main.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#include <uxr/agent/transport/tcp/TCPServerLinux.hpp>
+#endif //_WIN32
+
+int main(int argc, char** argv)
+{
+
+    eprosima::uxr::UDPServer* udp_server = new eprosima::uxr::UDPServer(2018);
+    eprosima::uxr::TCPServer* tcp_server = new eprosima::uxr::TCPServer(2018);
+#ifndef _WIN32
+    eprosima::uxr::SerialServer* serial_server = new eprosima::uxr::SerialServer(1, 0);
```
Comment (BorjaOuterelo) [inline test/cmake/main.cpp]: Where are the servers destroyed?.
May a smart pointer be useful here?.

Comment (julianbermudez) [reply test/cmake/main.cpp]: This exec should never been executed, but anyway you are right, I will change it.

Comment (julianbermudez) [reply test/cmake/main.cpp]: Fixed!

**Pull request merged**